apps/web/src/components/TopBar.tsx

```diff
@@ -58,65 +55,44 @@ export const TopBar: React.FC = () => {
-                          />
-                        )
-                      }
-                      colorScheme="none"
-                    />
-                  </Flex>
-                </button>
-              </PopoverTrigger>
-              <PopoverContent maxW="288px">
-                <PopoverBody borderRadius="6px" p="0">
-                  <Box pl="24px" pb="14px">
-                    <Text fontSize="18px" cursor="default" pt="18px">
-                      {user?.firstName && user?.lastName
-                        ? user.firstName + ' ' + user.lastName
-                        : 'Firstname Lastname'}
-                    </Text>
-                    <Text color="#888888" fontSize="18px" cursor="default">
-                      {userPosition && userPosition.name
-                        ? userPosition.name
-                        : 'Position'}
-                    </Text>
-                  </Box>
-                  <Divider />
-                  <button onClick={() => logout()}>
-                    <Flex align="center">
-                      <Text
-                        color="#4C658A"
-                        fontSize="18px"
-                        fontWeight="500"
-                        pl="24px"
-                        pr="12px"
-                        py="14px"
-                      >
-                        Sign out
-                      </Text>
-                      <SignoutIcon />
-                    </Flex>
-                  </button>
-                </PopoverBody>
-              </PopoverContent>
-            </>
-          )}
+          <PopoverTrigger>
+            <button>
+              <Flex align="center">
+                <IconButton
+                  aria-label="Visit profile"
+                  icon={
+                    user?.firstName && user?.lastName ? (
+                      <UserProfileAvatar
+                        firstName={user.firstName}
+                        lastName={user.lastName}
+                      />
+                    ) : (
+                      <UserProfileAvatar firstName="Default" lastName="User" />
+                    )
+                  }
+                  colorScheme="none"
+                />
+              </Flex>
+            </button>
+          </PopoverTrigger>
+          <PopoverContent maxW="288px">
+            <PopoverBody borderRadius="6px" p="0">
+              <Box pl="24px" pb="14px">
+                <Text fontSize="18px" cursor="default" pt="18px">
+                  {user?.firstName && user?.lastName
+                    ? user.firstName + ' ' + user.lastName
+                    : 'Firstname Lastname'}
+                </Text>
+                <Text color="#888888" fontSize="18px" cursor="default">
+                  {userPosition && userPosition.name
+                    ? userPosition.name
+                    : 'Position'}
+                </Text>
+              </Box>
+              <Divider />
+              <AzureSignout />
```

Comment:
I think that makes sense. I created SignIn, Signout, AzureSignin, and AzureSignout components that hopefully make the code cleaner. I think in the future we want to prioritize keeping logic and state in components rather than pages to promote modularity and separation of concerns. Just something to keep in mind for future reviews.

Reply:
Lit